bor/keeper_test.go

```diff
@@ -0,0 +1,172 @@
+package bor_test
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/maticnetwork/bor/common"
+	"github.com/maticnetwork/heimdall/app"
+	"github.com/maticnetwork/heimdall/bor"
+	bortypes "github.com/maticnetwork/heimdall/bor/types"
+	"github.com/maticnetwork/heimdall/checkpoint/simulation"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+type keeperTest struct {
+	suite.Suite
+
+	app *app.HeimdallApp
+	ctx sdk.Context
+}
+
+func TestBorKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(keeperTest))
+}
+
+func (suite *keeperTest) SetupTest() {
+	isCheckTx := false
+	suite.app = app.Setup(isCheckTx)
+	suite.ctx = suite.app.BaseApp.NewContext(isCheckTx, abci.Header{})
+}
+
+func (suite *keeperTest) TestFreeze() {
+	tc := []struct {
+		id, startBlock, endBlock uint64
+		seed                     common.Hash
+		borChainID               string
+		expErr                   error
+		msg                      string
+	}{
+		{
+			id:  1,
+			msg: "testing happy flow",
+		},
+		{
+			id:         1,
+			startBlock: 3,
+			msg:        "validation error missing id",
+		},
+	}
+
+	for i, c := range tc {
+		cMsg := fmt.Sprintf("i: %v, msg: %v", i, c.msg)
+		err := suite.app.BorKeeper.FreezeSet(suite.ctx, c.id, c.startBlock, c.endBlock, c.borChainID, c.seed)
+		suite.Equal(c.expErr, err, cMsg)
```

Comment:
Can we also check if span is stored or not, 
lastEthBlock is incremented or not.